sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -3406,42 +3406,1131 @@ lsc_format_ret(__ESIMD_NS::simd<T1, N> Vals) {
+__ESIMD_API std::enable_if_t<__ESIMD_DNS::get_num_args<Op>() == 0, simd<T, N>>
+slm_atomic_update(simd_view<simd<uint32_t, N>, RegionTy> byte_offset,
+                  simd_mask<N> mask = 1) {
+  return slm_atomic_update<Op, T, N>(byte_offset.read(), mask);
+}
+
+/// simd<T, N> atomic_update(local_accessor lacc,
+///                          simd<uint32_t, N> byte_offset,
+///                          simd_mask<N> pred = 1);
+///                                                             // (lacc-au0-1)
+/// Atomically updates \c N memory locations in SLM ssociated
+/// with the local accessor \p lacc at the given \p byte_offset,
+/// and returns a vector of old values found at the memory locations before
+/// update.
+template <atomic_op Op, typename T, int N, typename AccessorT>
+__ESIMD_API std::enable_if_t<
+    __ESIMD_DNS::get_num_args<Op>() == 0 &&
+        sycl::detail::acc_properties::is_local_accessor_v<AccessorT>,
+    simd<T, N>>
+atomic_update(AccessorT lacc, simd<uint32_t, N> byte_offset,
+              simd_mask<N> mask = 1) {
+  byte_offset += detail::localAccessorToOffset(lacc);
+  return slm_atomic_update<Op, T, N>(byte_offset, mask);
+}
+
+/// simd<T, N> atomic_update(local_accessor lacc,
+///                          simd_view<simd<uint32_t, N>, RegionTy> byte_offset,
+///                          simd_mask<N> pred = 1);
+///                                                             // (lacc-au0-2)
+/// Atomically updates \c N memory locations in SLM ssociated
+/// with the local accessor \p lacc at the given \p byte_offset,
+/// and returns a vector of old values found at the memory locations before
+/// update.
+template <atomic_op Op, typename T, int N, typename AccessorT,
+          typename RegionTy = region1d_t<uint32_t, N, 1>>
+__ESIMD_API std::enable_if_t<
+    __ESIMD_DNS::get_num_args<Op>() == 0 &&
+        sycl::detail::acc_properties::is_local_accessor_v<AccessorT>,
+    simd<T, N>>
+atomic_update(AccessorT lacc,
+              simd_view<simd<uint32_t, N>, RegionTy> byte_offset,
+              simd_mask<N> mask = 1) {
+  byte_offset += detail::localAccessorToOffset(lacc);
+  return slm_atomic_update<Op, T, N>(byte_offset, mask);
+}
+
+/// simd<T, N> atomic_update(local_accessor lacc,
+///                          simd_mask<N> pred = 1);
+///                                                             // (lacc-au0-3)
+/// Atomically updates \c N memory locations in SLM ssociated
+/// with the local accessor \p lacc at the given \p byte_offset,
+/// and returns a vector of old values found at the memory locations before
+/// update.
+template <atomic_op Op, typename T, int N, typename AccessorT>
+__ESIMD_API std::enable_if_t<
+    __ESIMD_DNS::get_num_args<Op>() == 0 &&
+        sycl::detail::acc_properties::is_local_accessor_v<AccessorT>,
+    simd<T, N>>
+atomic_update(AccessorT lacc, simd_mask<N> mask = 1) {
+  return slm_atomic_update<Op, T, N>(detail::localAccessorToOffset(lacc), mask);
+}
+
```

Comment:
This should be removed to simplify API and avoid undef behavior if operation is applied to the same address N times. It may even work for 'dec' (will it decrement N times).
But per our previous agreement (not adding scalar offset variants for gather/scatter/atomic_update), I suggest removing this one.
```suggestion
```

Reply:
Removed scalar offset overloads.